ui/app/routes/datasets/builder/DatasetSelector.tsx

```diff
@@ -128,56 +140,94 @@ export function DatasetSelector({
                           className="flex items-center justify-between"
                         >
                           <div className="flex items-center">
-                            <Plus className="mr-2 h-4 w-4 text-blue-500" />
-                            <span>
-                              {inputValue.trim() || "Start typing..."}
+                            <TablePlus
+                              className={clsx(
+                                "mr-2 h-4 w-4",
+                                inputValue.trim()
+                                  ? "text-blue-600"
+                                  : "text-fg-muted",
+                              )}
+                            />
+                            <span
+                              className={clsx(
+                                "text-sm",
+                                inputValue.trim()
+                                  ? "text-fg-primary font-mono"
+                                  : "text-fg-muted font-normal",
+                              )}
+                            >
+                              {inputValue.trim() ||
+                                "Start typing to create a new dataset..."}
```

Comment:
```suggestion
                              {inputValue.trim()}
```

Already conditionally renders this group only if `inputValue` is truthy